Automattic/wp-calypso
# Anchor: Get podcast title from API using podcast ID

#### Changes proposed in this Pull Request

* Moves over some Anchor-related hooks to get data, including the episode ID, and Spotify URL from the query params
* Auto-populate the site title field with the podcast title hook, which fetches from our API
* Wrap the `FlowRenderer` in a function component so we can use hooks to fetch data here.
* Update design selection to use the same podcast ID hook (temporarily -- eventually we'll want to set a field in the store to indicate a site is an Anchor site instead)

**Visuals**

<img width="914" alt="Screen Shot 2022-04-25 at 3 51 24 PM" src="https://user-images.githubusercontent.com/2124984/165164128-51e4fddb-4264-405a-8478-4bbb1997f908.png">

<img width="1390" alt="Screen Shot 2022-04-25 at 3 51 02 PM" src="https://user-images.githubusercontent.com/2124984/165164138-73a4a91b-37c4-4ac0-be6d-a60c6c90447f.png">

#### Testing instructions

* Switch to this PR, navigate to `/setup/?siteSlug=yoursite.wordpress.com&anchor_podcast=12345678` where `12345678` is your Anchor.fm podcast ID. You can find it by going to Podcast Availability, enabling RSS, and copying the ID from the RSS URL.
* Your podcast's title should be shown in the site title field when you land on the first screen instead of a placeholder.
* You should still be able to save and update your site title to whatever you want.
